velox/exec/Unnest.cpp

```diff
@@ -111,20 +112,61 @@ RowVectorPtr Unnest::getOutput() {
   }
 
   const auto size = input_->size();
-  const auto maxOutputSize = outputBatchRows();
 
   // Limit the number of input rows to keep output batch size within
-  // 'maxOutputSize' if possible. Process each input row fully. Do not break
-  // single row's output into multiple batches.
+  // 'maxOutputSize' if possible. Not process each input row fully when single
+  // row's output exceeds maxOutputSize. Single row's output maybe into
+  // multiple batches.
   vector_size_t numInput = 0;
   vector_size_t numElements = 0;
-  for (auto row = nextInputRow_; row < size; ++row) {
-    numElements += rawMaxSizes_[row];
+  vector_size_t partialProcessRowStartSize = -1;
```

Comment:
This is a lot of code to just add to an existing method. Perhaps, this logic can be extracted into a separate method or a few helper methods.